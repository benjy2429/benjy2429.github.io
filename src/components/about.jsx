import React from 'react';

export default () => (
  <section className="about container">
    <div className="row">
      <div className="col col-6 col-12-m">
        <h2>Work</h2>
        <p>
          I&#8217;m currently working as a software developer for{' '}
          <a href="https://www.sky.com">Sky</a> in Leeds.
        </p>
        <p>
          I joined through the Software Engineering Academy graduate scheme.
          This involved a six month project developing a piece of software in an
          agile team. In our team, we created a digital whiteboard for
          visualising and orgainising development squads.
        </p>
        <p>
          These days you&#8217;ll find me in the Digital Help team programming
          mainly in ReactJS and Ruby. I deal with frontend apps such as the{' '}
          <a href="https://www.sky.com/help">Help Homepage</a> and{' '}
          <a href="https://contactus.sky.com">Contact Us</a> as well as backend
          services and CMS&#8217;s that power the help articles and diagnostic
          flows.
        </p>
      </div>
      <div className="col col-6 col-12-m">
        <h2>Education</h2>
        <p>
          I graduated from{' '}
          <a href="https://www.sheffield.ac.uk/">The University of Sheffield</a>{' '}
          in 2015 with a First Class Masters degree in Computer Science. For my
          dissertation, I developed a real-time shadow puppet storytelling
          application using Microsoftâ€™s Kinect sensor and C++.
        </p>
        <p>
          I also worked as a Student Abassador during my final years. This
          involved helping out at open days and events, demonstrating in
          technical activities and giving presentations about life as a student.
        </p>
      </div>
    </div>
  </section>
);
