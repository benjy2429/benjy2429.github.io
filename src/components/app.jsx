import React from 'react';
import dataLoader from './data-loader';
import Header from './header';
import About from './about';
import ProjectsWrapper from './projects/projects-wrapper';
import Contact from './contact';
import Footer from './footer/footer';

export const App = ({ projects, footer }) => (
  <React.Fragment>
    <Header />
    <About />
    <ProjectsWrapper projects={projects} />
    <Contact />
    <Footer {...footer} />
  </React.Fragment>
);

export default dataLoader(App);
