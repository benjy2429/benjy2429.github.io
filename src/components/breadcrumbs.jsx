import React from 'react';
import Link from 'gatsby-link';

export default ({ crumbs = [] }) =>
  !crumbs.length ? null : (
    <ul className="breadcrumbs">
      <li className="breadcrumb">
        <Link to="/">Home</Link>
      </li>
      {crumbs.map(({ name, href }) => (
        <li key={`crumb-${name}`} className="breadcrumb">
          {href ? <Link to={href}>{name}</Link> : name}
        </li>
      ))}
    </ul>
  );
