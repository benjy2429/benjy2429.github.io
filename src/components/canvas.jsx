import React from 'react';
import Delaunator from 'delaunator';

const POINT_COUNT = 20;
const TRIANGLE_ALPHA = 0.05;

const randomPoint = (width, height) => {
  const x = Math.random() * width;
  const y = Math.random() * height;
  return [x, y];
};

const drawTriangle = (ctx, points) => {
  ctx.beginPath();
  ctx.moveTo(points[0][0], points[0][1]);
  ctx.lineTo(points[1][0], points[1][1]);
  ctx.lineTo(points[2][0], points[2][1]);
  ctx.closePath();

  const alpha = Math.random() * TRIANGLE_ALPHA;
  ctx.fillStyle = `rgba(0, 0, 0, ${alpha})`;
  ctx.fill();
};

const drawBackground = (ctx, width, height) => {
  const points = Array(POINT_COUNT)
    .fill(0)
    .map(() => randomPoint(width, height));

  points.push([0, 0]);
  points.push([0, height]);
  points.push([width, 0]);
  points.push([width, height]);

  const { triangles } = Delaunator.from(points);

  for (let i = 0; i < triangles.length; i += 3) {
    const p0 = triangles[i];
    const p1 = triangles[i + 1];
    const p2 = triangles[i + 2];
    drawTriangle(ctx, [points[p0], points[p1], points[p2]]);
  }
};

export default class Canvas extends React.Component {
  constructor(props) {
    super(props);
    this.canvas = React.createRef();
  }

  componentDidMount() {
    const canvas = this.canvas.current;
    const ctx = canvas.getContext('2d');

    canvas.width = canvas.offsetWidth;
    canvas.height = canvas.offsetHeight;

    drawBackground(ctx, canvas.width, canvas.height);
  }

  render() {
    return <canvas ref={this.canvas} id="background-canvas" />;
  }
}
