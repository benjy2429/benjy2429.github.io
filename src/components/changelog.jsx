import React from 'react';
import moment from 'moment';

export default ({ items = [] }) => (
  <div>
    <h1>Changelog</h1>

    {items.map(({ version, date, description }) => {
      const parsedDate = moment(date);
      return (
        <div className="changelog-item" key={version}>
          <p>
            <span className="changelog-version">{version}</span>
            <br />
            {parsedDate.isValid() ? (
              <span
                className="changelog-date"
                title={parsedDate.format('Do MMM YYYY')}
              >
                {parsedDate.fromNow()}
              </span>
            ) : (
              <span className="changelog-date">TBC</span>
            )}
          </p>
          <p className="changelog-description">{description}</p>
        </div>
      );
    })}
  </div>
);
