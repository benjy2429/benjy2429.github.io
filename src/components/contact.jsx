import React from 'react';

export default () => (
  <section className="contact">
    <div className="container">
      <h2 id="contact">Contact</h2>
      <p>
        I hope I&#8217;ve covered everything but if you have any questions or
        comments, drop me a message.
      </p>

      <form action="https://www.enformed.io/6b8u88yv" method="POST">
        <input
          type="hidden"
          name="*redirect"
          value="https://www.bencarr.co.uk/thanks"
        />
        <input
          type="hidden"
          name="*subject"
          value="bencarr.co.uk Contact Form"
        />
        <input type="hidden" name="*reply" value="email" />
        <input type="text" name="*honeypot" style={{ display: 'none' }} />

        <label htmlFor="contact-name">Name</label>
        <input type="text" name="name" id="contact-name" required />

        <label htmlFor="contact-email">Email</label>
        <input type="email" name="email" id="contact-email" required />

        <label htmlFor="contact-message">Message</label>
        <textarea name="message" id="contact-message" required />

        <input type="submit" value="Send" />
      </form>
    </div>
  </section>
);
