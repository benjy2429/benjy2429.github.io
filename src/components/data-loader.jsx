import React from 'react';
import data from '../data.json';

export default (WrappedComponent, _data = data) =>
  class extends React.Component {
    constructor(props) {
      super(props);
      this.data = _data;
    }
    render() {
      return <WrappedComponent {...this.props} {...this.data} />;
    }
  };
