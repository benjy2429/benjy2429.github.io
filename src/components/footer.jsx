import React from 'react';
import converter from 'number-to-words';

const yearInWords = () => {
  const year = new Date().getFullYear().toString();
  const first = year.slice(0, 2);
  const second = year.slice(2, 4);
  return `${converter.toWords(first)}-${converter.toWords(second)}`;
};

export default ({ socialLinks }) => (
  <footer className="footer">
    <div className="container">
      <p className="footer-title">
        Ben Carr
        <br />
        Software Developer
      </p>

      <div className="footer-grid">
        <div className="footer-grid-left">
          <h3 className="footer-links-title">Menu</h3>
          <ul className="footer-links">
            <li>
              <a href="#work" className="link-light">
                Work
              </a>
            </li>
            <li>
              <a href="#education" className="link-light">
                Education
              </a>
            </li>
            <li>
              <a href="#projects" className="link-light">
                Projects
              </a>
            </li>
            <li>
              <a href="#contact" className="link-light">
                Contact
              </a>
            </li>
          </ul>
        </div>

        <div className="footer-grid-left">
          <h3 className="footer-links-title">Social</h3>
          <ul className="footer-links">
            {socialLinks.map(link => (
              <li key={link.name}>
                <a href={link.href} className="link-light">
                  {link.name}
                </a>
              </li>
            ))}
          </ul>
        </div>

        <div className="footer-grid-right">
          Copyright &copy; {yearInWords()}
          {false && (
            <React.Fragment>
              <br />
              v1.1-alpha
              <br />
              <a href="/">Changelog</a>
            </React.Fragment>
          )}
        </div>
      </div>
    </div>
  </footer>
);
