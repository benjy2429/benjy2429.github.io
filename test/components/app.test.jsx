import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { App } from '../../src/components/app';

const props = {
  projects: 'projects',
  socialLinks: 'socialLinks'
};

describe('App component', () => {
  it('renders correctly', () => {
    const component = shallow(<App {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
