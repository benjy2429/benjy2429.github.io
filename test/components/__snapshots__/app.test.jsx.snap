// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<React.Fragment>
  <Component />
  <Component />
  <Component
    projects="projects"
  />
  <Component />
  <Component
    footerProps="footer"
  />
</React.Fragment>
`;
